driver/src/test/unit/com/mongodb/client/gridfs/GridFSBucketSpecification.groovy

```diff
@@ -160,16 +151,22 @@ class GridFSBucketSpecification extends Specification {
 
     def 'should upload from stream'() {
         given:
+        def findIterable = Mock(FindIterable)
         def filesCollection = Mock(MongoCollection)
         def chunksCollection = Mock(MongoCollection)
-        def gridFSBucket = new GridFSBucketImpl(Stub(MongoDatabase), 'fs', 255, Stub(CodecRegistry), Stub(ReadPreference),
-                Stub(WriteConcern), readConcern, filesCollection, chunksCollection, true)
+        def gridFSBucket = new GridFSBucketImpl('fs', 255, filesCollection, chunksCollection)
         def contentBytes = 'content' as byte[]
         def inputStream = new ByteArrayInputStream(contentBytes)
 
         when:
         gridFSBucket.uploadFromStream('filename', inputStream)
 
+        then: 'index check'
```

Comment:
Thats the check to see if a file exists, if not does the index exists check if that fails it adds the index.  This is just doing the checking the file exists part.